sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/rest_client.hpp

```diff
@@ -2200,6 +2207,218 @@ namespace Azure { namespace Storage { namespace Blobs {
+      AZ_STORAGE_BLOBS_DLLEXPORT const static BlobQueryArrowFieldType Int64;
+      AZ_STORAGE_BLOBS_DLLEXPORT const static BlobQueryArrowFieldType Bool;
+      AZ_STORAGE_BLOBS_DLLEXPORT const static BlobQueryArrowFieldType Timestamp;
+      AZ_STORAGE_BLOBS_DLLEXPORT const static BlobQueryArrowFieldType String;
+      AZ_STORAGE_BLOBS_DLLEXPORT const static BlobQueryArrowFieldType Double;
+      AZ_STORAGE_BLOBS_DLLEXPORT const static BlobQueryArrowFieldType Decimal;
+
+    private:
+      std::string m_value;
+    };
+    /**
+     * @brief Groups settings regarding specific field of an arrow schema.
+     */
+    struct BlobQueryArrowField final
+    {
+      /**
+       * Type of blob query arrow field.
+       */
+      BlobQueryArrowFieldType Type;
+      Nullable<std::string> Name;
+      Nullable<int32_t> Precision;
+      Nullable<int32_t> Scale;
+    };
+    namespace _detail {
+      /**
+       * @brief Groups the settings used for formatting the response if the response should be Arrow
+       * formatted.
+       */
+      struct ArrowConfiguration final
+      {
+        /**
+         * Array of BlobQueryArrowField.
+         */
+        std::vector<BlobQueryArrowField> Schema;
+      };
+      /**
+       * @brief Parquet configuration.
+       */
+      struct ParquetConfiguration final
+      {
+      };
+      struct QueryFormat final
+      {
+        /**
+         * The quick query format type.
+         */
+        QueryFormatType Type;
+        /**
+         * Groups the settings used for interpreting the blob data if the blob is delimited text
+         * formatted.
+         */
+        Nullable<_detail::DelimitedTextConfiguration> DelimitedTextConfiguration;
+        /**
+         * Json text configuration.
+         */
+        Nullable<_detail::JsonTextConfiguration> JsonTextConfiguration;
+        /**
+         * Groups the settings used for formatting the response if the response should be Arrow
+         * formatted.
+         */
+        Nullable<_detail::ArrowConfiguration> ArrowConfiguration;
+        /**
+         * Parquet configuration.
+         */
+        Nullable<ParquetConfiguration> ParquetTextConfiguration;
+      };
+      struct QuerySerialization final
+      {
+        QueryFormat Format;
+      };
+      /**
+       * @brief Groups the set of query request settings.
+       */
+      struct QueryRequest final
+      {
+        /**
+         * Required. The type of the provided query expression.
+         */
+        QueryRequestQueryType QueryType;
+        /**
+         * The query expression in SQL. The maximum size of the query expression is 256KiB.
+         */
+        std::string Expression;
+        Nullable<QuerySerialization> InputSerialization;
+        Nullable<QuerySerialization> OutputSerialization;
+      };
+    } // namespace _detail
+    /**
+     * @brief Response type for #Azure::Storage::Blobs::BlobClient::Query.
+     */
+    struct QueryBlobResult final
```

Comment:
Seems to be missing some fields:
Content-Encoding
Content-Language
Cache-Control
Content-Disposition
Access-Control-Allow-Origin
Access-Control-Expose-Headers
Vary
Access-Control-Allow-Credentials
x-ms-blob-committed-block-count
x-ms-server-encrypted

Reply:
Where did you get these?
In .Net, they just reuse the struct for `Download()`. But apparently that's wrong.
I've tested, storage service doesn't return all those headers as stated in swagger file/API reference.

So the situation is, we don't know which headers will be returned and which won't. I don't think customers will expect to get those headers from `Query()` API anyway.

Reply:
Got it from the official doc:
https://docs.microsoft.com/en-us/rest/api/storageservices/query-blob-contents
Might be a doc issue, please double check.

Reply:
According to the doc, most of the headers listed above only return under a certain condition. I am fine with you adding it later but from a feature completeness perspective, it would be nice to have them in version 1.

Reply:
I decided to hold it off until we have customer request.